Adding a Nordic layout for UT47

NeonSpork opened the pull request:
Hello,

I humbly request that my Nordic layout for the UT47 be merged with the main repository. I've played around with it and made some changes I think would be valuable to other Nordic UT47 users out there. 

A basic description of the layout, in addition to a somewhat large .png file with an overview of the keymap is included in the readme.md file.

As a side note: My experience with keymap files is quite limited, and there is a chance that my code can be trimmed (especially at the top in the definitions), any suggestions are welcome. Thank you for your time!

drashna commented on keyboards/ut47/keymaps/Nordic/keymap.c:
```diff
@@ -0,0 +1,277 @@
+#include QMK_KEYBOARD_H
+#ifdef LED_ENABLE
+  #include "protocol/serial.h"
+
```
Should be an `#endif` here.

NeonSpork replied:
Sorted this.

drashna commented on keyboards/ut47/keymaps/Nordic/keymap.c:
```diff
@@ -0,0 +1,277 @@
+#ifdef LED_ENABLE
+  #include "protocol/serial.h"
+
+#ifndef KEYMAP_NORDIC_H
```
You shouldn't need all of this.  A simple `#include "keymap_nordic.h"` should handle all of this. 

If not, then it may be worth creating a `keymap_nordic.h` in the keymap folder, and then use that, until the main nordic file can be changed.

NeonSpork commented:
Hi, I re-uploaded the files to a new folder in lower case called "nordic". I also added the #endif you requested and changed to #include "keymap_nordic.h".

drashna commented:
@NeonSpork If you include the keymap_nordic.h file, you shouldn't need all the additional defines.  

Eg, this block: https://github.com/NeonSpork/qmk_firmware/blob/a53b155c826f78cd300ca15eed2fc6854ec3f883/keyboards/ut47/keymaps/nordic/keymap.c#L26-L119

And sorry, my bad, you'd want the `#include "keymap_norwegian.h"` rather than `#include "keymap_nordic.h"`


If you want to keep them in there, that should be okay, but you're basically redefining stuff.

NeonSpork commented:
Oh I see what you mean! I'll edit that out in that case, no need for redundancy. I'm learning as I go here :)

drashna commented:
It's not a problem, that's what I'm here for. :)
And same! I'm still learning stuff!

NeonSpork commented:
Cleaned up the definitions in the beginning of the keymap file as suggested.

drashna commented:
Awesome, thanks!

NeonSpork commented:
I forgot to add an include (#include "keymap_nordic.h"). Not sure if it's fully necessary, but think it may be to make sure the nordic specialty keys work ok. Do I need to create a newpull request to add the changes?

drashna commented:
It's not.  The `keymap_norwegian.h` file actually includes it, so by including that one, you include both. 

To show you: 
https://github.com/qmk/qmk_firmware/blob/master/quantum/keymap_extras/keymap_norwegian.h#L19

So that you got that means that you have everything you should need.  Which is why I mentioned that you wanted that one, above. 

So you should be good.

NeonSpork commented:
Oh perfect! I'll leave it alone then. Thanks for the help :)

drashna commented:
You're very welcome!
